### PR TITLE
[WorksForYou] Wake app and preload when notification received. 📱

### LINKED ISSUE #1119: [Push Notifications] Make app launch in background, so we can record ‘received’ event
Not sure if this is allowed or not, but we can use the `content-available` flag to indicate that the app should launch in the background (to prefetch data).

Right now the ‘received’ vs ‘tapped’ events are without distinction. Receiving a notification does not necessarily trigger the app to load, so all of those ‘received’ notifications were people that either tapped, or had the app open at the time they received it.

### PR DESCRIPTION
**Pull request by alloy**
Fixes #1119 and #595.

In short, I added a ‘host’ VC which holds the real Works For You VC. It creates a new instance of the Works For You VC when:
- It receives the `reloadData` message, such as when a notification is received.
- An hour has passed.

The reasons for doing this are:
- Make it possible to preload the Works For You VC.
- Fixes issue with bell tab not getting a unread count badge when a notification is received.

### CODE REVIEW

**Comment by orta** [comment]
I know this GIF

File: Artsy/App/ARAppNotificationsDelegate.m [hunk 1]
```diff
@@ -102,21 +96,35 @@ - (void)applicationDidReceiveRemoteNotification:(NSDictionary *)userInfo inAppli
...
+
+    } else if (applicationState == UIApplicationStateInactive) {
+        // The user tapped a notification while the app was in background.
         [self tappedNotification:notificationInfo viewController:viewController];
     }
 }
 
+- (void)receivedNotification:(NSDictionary *)notificationInfo viewController:(UIViewController *)viewController;
+{
+    [ARAnalytics event:ARAnalyticsNotificationReceived withProperties:notificationInfo];
```
Comment (orta) [inline Artsy/App/ARAppNotificationsDelegate.m]: should this go to the DSL?

File: Artsy/App/AROptions.m [hunk 2]
```diff
@@ -17,8 +16,7 @@ + (NSArray *)labsOptions
         AROptionsUseVCR,
         AROptionsSettingsMenu,
         AROptionsTappingPartnerSendsToPartner,
-        AROptionsUseNativeAuctions,
-        AROptionsUseNativeWorksForYou
```
Comment (orta) [inline Artsy/App/AROptions.m]: cool

File: Artsy/View_Controllers/WorksForYou/ARWorksForYouReloadingHostViewController.m [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    self.lastLoadedAt = [NSDate date];
+}
+
+- (void)setWorksForYouViewController:(ARWorksForYouViewController *)viewController;
+{
+    if (_worksForYouViewController != viewController) {
+        if (_worksForYouViewController) {
+            [_worksForYouViewController willMoveToParentViewController:nil];
+            [_worksForYouViewController.view removeFromSuperview];
+            [_worksForYouViewController removeFromParentViewController];
```
Comment (orta) [inline Artsy/View_Controllers/WorksForYou/ARWorksForYouReloadingHostViewController.m]: `ar_addChildViewController` ?

File: Artsy/View_Controllers/WorksForYou/ARWorksForYouReloadingHostViewController.m [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        [self addWorksForYouViewController];
+    }
+}
+
+- (void)addWorksForYouViewController;
+{
+    [self addChildViewController:self.worksForYouViewController];
+    [self.view addSubview:self.worksForYouViewController.view];
+    [self.worksForYouViewController.view alignToView:self.view];
+    [self.worksForYouViewController didMoveToParentViewController:self];
```
Comment (orta) [inline Artsy/View_Controllers/WorksForYou/ARWorksForYouReloadingHostViewController.m]: and I think there's `ar_addAlignedViewController` for this

**Comment by alloy** [comment]
> I know this GIF

This needs more discussion.

File: Artsy/App/ARAppNotificationsDelegate.m (hunk 1 above)
Comment (alloy) [inline Artsy/App/ARAppNotificationsDelegate.m]: Hmm, good question, will take a look at the required (test) changes for that.

**Comment by ArtsyMobile** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%">1 Warning</th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:warning:</td>
      <td><h4>Found unstubbbed networking requests</h4>

<ul>
<li><a href="https://github.com/search?q=ARFairFavoritesNetworkModel+repo%3Aartsy%2Feigen&amp;ref=searchresults&amp;type=Code&amp;utf8=%E2%9C%93">ARFairFavoritesNetworkModelSpec</a> in <code>test_ignores_artworks_without_a_partner</code></li>
<li><a href="https://github.com/search?q=ARFairFavoritesNetworkModel+repo%3Aartsy%2Feigen&amp;ref=searchresults&amp;type=Code&amp;utf8=%E2%9C%93">ARFairFavoritesNetworkModelSpec</a> in <code>test_ignores_artworks_without_a_partner</code></li>
<li><a href="https://github.com/search?q=ARFairFavoritesNetworkModel+repo%3Aartsy%2Feigen&amp;ref=searchresults&amp;type=Code&amp;utf8=%E2%9C%93">ARFairFavoritesNetworkModelSpec</a> in <code>test_ignores_artworks_without_a_partner</code></li>
</ul>
</td>
    </tr>
    
  </tbody>
</table>
  

  


<p align="right" data-meta="generated_by_danger" data-base-commit="" data-head-commit="" >
  Generated by :no_entry_sign: <a href="https://github.com/danger/danger/">danger</a>
</p>

File: Artsy_Tests/Stubs/ARStubbedWorksForYouNetworkModel.m [hunk 5]
```diff
@@ -0,0 +1,72 @@
+#import "ARStubbedWorksForYouNetworkModel.h"
+
+#import "Artist.h"
+#import "Artwork.h"
+#import "Artwork+Extensions.h"
+
+@implementation ARStubbedWorksForYouNetworkModel
```
Comment (sarahscott) [inline Artsy_Tests/Stubs/ARStubbedWorksForYouNetworkModel.m]: :tada: Love this - my WorksForYouVC test file was getting crowded

**Comment by sarahscott** [comment]
Agreed re: the gif issue. Otherwise, this looks awesome. Merging!

File: Artsy_Tests/Stubs/ARStubbedWorksForYouNetworkModel.m (hunk 5 above)
Comment (alloy) [inline Artsy_Tests/Stubs/ARStubbedWorksForYouNetworkModel.m]: :taco:

**Pull request merged**